sympy/solvers/diophantine.py

```diff
@@ -152,11 +152,24 @@ def diophantine(eq, param=symbols("t", integer=True), syms=None):
             if not is_sequence(syms):
                 raise TypeError(
                     'syms should be given as a sequence, e.g. a list')
-            syms = [i for i in syms if i in var]
+
             if syms != var:
```

Comment:
I added the [line](https://github.com/sympy/sympy/pull/11239/files#diff-73aa5ec817dd875d07ed6675b9b63574R155) . Thanks for review.